apiserver/registration.go

```diff
@@ -172,17 +169,18 @@ func (h *registerUserHandler) processPost(
 	return userTag, response, nil
 }
 
-func getConnectorInfoer(ctx context.Context, model stateenvirons.Model, cloudService common.CloudService, credentialService common.CredentialService) (environs.ConnectorInfo, error) {
-	configGetter := stateenvirons.EnvironConfigGetter{
-		Model: model, CloudService: cloudService, CredentialService: credentialService}
-	environ, err := common.EnvironFuncForModel(model, cloudService, credentialService, configGetter)(ctx)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	if connInfo, ok := environ.(environs.ConnectorInfo); ok {
-		return connInfo, nil
+func getConnectorInfoer(
+	ctx context.Context,
+	providerFactory providertracker.ProviderFactory,
+	controllerModelID model.UUID,
+) (environs.ConnectorInfo, error) {
+
+	type connectableProvider interface {
+		environs.BootstrapEnviron
+		environs.ConnectorInfo
 	}
-	return nil, errors.NotSupportedf("environ %q", environ.Config().Type())
+	return providertracker.ProviderRunner[connectableProvider](
```

Comment:
This chunk of code is pretty obscure to me. It could do with some comments at least on the interface type there, and the function here could be refactored to be given a variable name, or at least have the bracketing over another line to make it clear you are getting a function back then imidietly calling it with the ctx.